src/renderer/main.js

```diff
@@ -48,7 +48,14 @@ module.exports.api = (windowId) => {
     const mainWindowOptions = windowOptionSet.options || {};
     const enableV2Api = (mainWindowOptions.experimental || {}).v2Api;
     const v2AdapterShim = (!enableV2Api ? '' : jsAdapterV2);
-    const optionsString = JSON.stringify(windowOptionSet);
 
-    return `global.__startOptions = ${optionsString}; ${me} ; ${jsAdapter}; ${v2AdapterShim} ; fin.__internal_.ipc = null;`;
+    windowOptionSet.runtimeArguments = JSON.stringify(coreState.args);
```

Comment:
This is **temporary** and needed for the api-decorator and javascript-adapter to conditionally add/change new functionality based on the presence of `--async-plugins` flag in runtime arguments